eng/tools/openapi-diff-runner/src/utils/spec.ts

```diff
@@ -155,3 +157,20 @@ export function getBaseNameForSwagger(filePath: string, version: string = ""): s
   }
   return basename(filePath);
 }
+
+/**
+ * Deduplicate swagger objects in an array by their path.
+ * @param swaggers The array of swagger objects to deduplicate.
+ * @returns A new array with duplicate swagger objects removed.
+ */
+export function deduplicateSwaggers(swaggers: Swagger[]): Swagger[] {
+  // Deduplicate by path
+  const seen = new Set();
+  return swaggers.filter((swagger) => {
+    if (seen.has(swagger.path)) {
+      return false;
+    }
+    seen.add(swagger.path);
+    return true;
+  });
```

Comment:
```suggestion
  return (new Map(swaggers.map(s => [s.path, s])).values();
```

I think this is equivalent, with the small caveat `new Map()` uses last-wins, while your code uses first-wins, but I don't think you care.